src/convert-plantuml.js

```diff
@@ -4,44 +4,94 @@ const fs = require('fs');
-    .option('-i, --image-path [imagePath]', 'Save path for the generated images')
-    .option('-p, --image-prefix [imagePrefix]', 'A prefix to add to generated image filenames')
-    .option('-o, --output-file [outputFile]', 'output file to write converted markdown. Default is empty' )
-    .action(function(markdownFile) {
-      const readFile = () => {
-        return fs.readFileSync(markdownFile, 'utf-8')
-            .toString();
-      };
-      const markdown = readFile();
-      const options = this.opts();
-      const extension = options.extension;
-      const prefix = options.imagePrefix;
-      const imagePath = options.imagePath;
-      const markdownFilePath = options.markdownFilePath;
-      const pumlMdSnippetMatch = MdPumlMatchers.resetAndGetMarkdownPlantumlSnippetMatcher();
-      const processedOutput = markdown.replace(pumlMdSnippetMatch,
-          (s) => convertPumlMarkup(s, extension, markdownFilePath, prefix, imagePath));
-      const outputFile = options.outputFile ? options.outputFile : markdownFile;
-      fs.writeFileSync(outputFile, processedOutput);
-      process.exit();
-    });
-
-
-mdPumlConverterCommand.parse();
+
+const initCommandAndParse = () => {
+  const mdPumlConverterCommand = new Command();
+  let parsedConversionParams;
+  mdPumlConverterCommand
+      .command('run')
+      .argument('<markdownFile>', 'The markdown file which contains plantuml markups\n' +
+            'in the following form ```plantuml \n ....```')
+      .requiredOption('-e, --extension <extension>', 'Image type extension e.g.: \'svg\', \'png\' (default), \'jpg\'')
+      .requiredOption('-f, --markdown-file-path <markdownFilePath>', 'Relative path of the current markdown file')
+      .option('-i, --image-path [imagePath]', 'Save path for the generated images')
+      .option('-p, --image-prefix [imagePrefix]', 'A prefix to add to generated image filenames')
+      .option('-o, --output-file [outputFile]', 'output file to write converted markdown. Default is empty' )
+      .action(function(markdownFile) {
+        const markdown = readFile(markdownFile);
+        const options = this.opts();
+        const extension = options.extension;
+        const prefix = options.imagePrefix;
+        const imagePath = options.imagePath;
+        const markdownFilePath = options.markdownFilePath;
+        const outputFile = options.outputFile ? options.outputFile : markdownFile;
+        parsedConversionParams = new ConversionParams(
+            markdown,
+            extension,
+            prefix,
+            imagePath,
+            markdownFilePath,
+            outputFile,
+        );
+      });
+  mdPumlConverterCommand.parse();
+  return parsedConversionParams;
+};
+
+const main = async () => {
+  const parsedConversionParams = initCommandAndParse();
+  const processedOutput = processMarkdown(parsedConversionParams);
+  fs.writeFileSync(parsedConversionParams.outputFile, processedOutput);
```

Comment:
Might as well use await fs.writeFile here since you're already in async context

Reply:
will do

Reply:
done